database/rrdhost.c

```diff
@@ -473,6 +460,30 @@ int is_legacy = 1;
         host->system_info->mc_version = enable_metric_correlations ? metric_correlations_version : 0;
     }
 
+    // ------------------------------------------------------------------------
+    // link it and add it to the index
+
+    rrd_wrlock();
+
+    RRDHOST *t = rrdhost_index_add_by_guid(host);
+    if(t != host) {
+        error("Host '%s': cannot add host with machine guid '%s' to index. It already exists as host '%s' with machine guid '%s'.", rrdhost_hostname(host), host->machine_guid, rrdhost_hostname(t), t->machine_guid);
```

Comment:
Hello @ktsaou ,

I am far from to reach this point of the code, but I am already pasting some covertiy warnings for you evaluate:

```sh
** CID 433491:    (SLEEP)
/database/rrdhost.c: 476 in rrdhost_create()
470             error("Host '%s': cannot add host with machine guid '%s' to index. It already exists as host '%s' with machine guid '%s'.", rrdhost_hostname(host), host->machine_guid, rrdhost_hostname(t), t->machine_guid);
471             rrdhost_free___while_having_rrd_wrlock(host, true);
472             rrd_unlock();
473             return NULL;
474         }
475     
>>>     CID 433491:    (SLEEP)
>>>     Call to "rrdhost_index_add_hostname" might sleep while holding lock "rrd_rwlock.rwlock_t".
476         rrdhost_index_add_hostname(host);
477     
478         if(is_localhost)
479             DOUBLE_LINKED_LIST_PREPEND_UNSAFE(localhost, host, prev, next);
480         else
481             DOUBLE_LINKED_LIST_APPEND_UNSAFE(localhost, host, prev, next);
```

Reply:
Due to spinlock. Ignore it.